docs/build/isc/v1.1/docs/tutorials/defi-yield-farming.md

```diff
@@ -0,0 +1,422 @@
+    const TokenFarm = await ethers.getContractFactory("TokenFarm");
+
+    /// Deploy contracts
+    const dappToken = await DappToken.deploy();
+    await dappToken.deployed();
+    console.log("DappToken deployed to:", dappToken.address);
+
+    const iotaToken = await IOTAToken.deploy();
+    await iotaToken.deployed();
+    console.log("IOTAToken deployed to:", iotaToken.address);
+
+    const tokenFarm = await TokenFarm.deploy(dappToken.address, iotaToken.address, "PYTH_ORACLE_CONTRACT_ADDRESS");
+    await tokenFarm.deployed();
+    console.log("TokenFarm deployed to:", tokenFarm.address);
+}
+
+main()
+    .then(() => process.exit(0))
+    .catch((error) => {
+        console.error(error);
+        process.exit(1);
+    });
+
+```
+
+## Step 4: Deploying to the IOTA EVM Test
+
+### 1. Configure Hardhat for IOTA EVM Testnet
+
+Create a `.env` file with the following content (replace with your own private key and RPC URL):
+
+```bash
+PRIVATE_KEY=your-private-key
+IOTA_RPC_URL=https://json-rpc.evm.testnet.iotaledger.net
+```
+
+### 2. Install the dotenv package to manage environment variables:
+
+```bash
+npm install dotenv
+```
+
+### 3. Import the `dotenv` package in your `hardhat.config.js`:
+
+Modify `hardhat.config.js` to include the IOTA EVM testnet configuration:
+
+```javascript
+require("@nomicfoundation/hardhat-toolbox");
+require("dotenv").config();
+
+module.exports = {
+    solidity: "0.8.7",
+    networks: {
+        'iotaevm-testnet': {
+            url: process.env.IOTA_RPC_URL,,
+            chainId: 1075,
+            accounts: [process.env.PRIVATE_KEY], // Ensure your .env file has PRIVATE_KEY set
+        },
+    },
+};
+```
+
+## Step 5: Deploy to IOTA EVM Testnet
+
+<DeployAdmonition />
+
+### 1. Compile your contracts:
+
+```bash
+npx hardhat compile
+```
+
+### 2. Deploy your contracts to the IOTA EVM testnet:
+
+```bash
+npx hardhat run scripts/deploy.js --network evm-testnet
+```
+
+You can verify your contract by visiting
+the [EVM Testnet Explorer](https://explorer.evm.testnet.shimmer.network/),
+and searching for the address from the previous step. If you access the `Contract` tab, you should be able to see your code and interact with your contract.
+
+Feel free to [fork this repository](https://github.com/iota-community/Defi-yield-farming), contribute with pull requests, or customize it to suit your needs. We have a React [frontend](https://github.com/iota-community/Defi-yield-farming/tree/main/frontend) on the repository that you can use to interact with Token Farm smart contract.
+
+### Key Concepts Covered:
+
+1. ERC20 Tokens: Both the reward and staking tokens are standard ERC20 tokens.
+2. Staking Mechanism: Users can deposit (stake) IOTATokens into the TokenFarm contract.
+3. Unstaking Mechanism: Users can withdraw (unstake) their staked tokens at any time.
+4. Reward Distribution: The contract periodically distributes DappTokens as rewards to stakers.
```

Comment:
I think something like this would be more useful at the beginning so the users know what the tutorial will cover.